_posts/2019-02-30-nlog-4-6-is-live.md

```diff
@@ -0,0 +1,226 @@
+        </filters>
+    </logger>
+</rules>
+```
+
+In NLog 5 the default will be probably changed to "ignore"
+
+### Substring, left and right
+
+New layout renders has been added:
+
+```
+${substring:${level}:start=2:length=2}    // from index 2, length 2
+${substring:${level}:start=-2:length=2}   // from end -2, length 2
+${left:${level}:length=2}    // from start, length 2
+${right:${level}:length=2}   // from end, length 2
+```
+
+### Name filter: support for multiple wildcards and in any position
+In NLog 4.5 and before, the wildcards in the name filter of the <rule> element was limited. 
+
+- `?` was not supported
+- Only one `*` was allowed the name filter. 
+
+With a new implementation, those restrictions are now gone and `?` wildcard has been added.
+
+- `*` - matches 0 or more characters
+- `?` - matches exactly 1 character
+
+Examples
+
+```xml
+  <logger name="*.Library.*" minlevel="Warn" writeTo="f3" />
+  <logger name="*TcpTestServer[*].Connection[??].*" writeTo="logconsole" />
+```
+
+
+### InternalLogFile variables
+
+The internalLogFile attribute is for stability limited in features, no layout renderers where allowed. 
+
+In NLog 4.6 we added support for: ${currentdir}, ${basedir} & ${tempdir}, and for environment variables, like `%myPath%` 
+
+example:
+
+```xml
+<nlog internalLogFile="${basedir}\log.txt" internalLogLevel="Trace">
+   <targets>
+      <!-- target configuration here -->
+   </targets>
+   <rules>
+      <!-- log routing rules -->
+   </rules>
+</nlog>
+```
+
+
+### AsyncWrapper uses ConcurrentQueue
+ConcurrentQueue has been optimized with .NET Standard 2.0, so it provides high performance and little allocation.
+
+NLog AsyncWrapper will now automatically make use of ConcurrentQueue when running on .NET Core 2 platform.
+This will greatly reduce the congestion overhead of having multiple threads writing to the same AsyncWrapper.
+
+### AsyncWrapper with faster timer
+AsyncWrapper no longer schedules a timer every 50 ms even if nothing to write. Instead it will schedule a single
+timer on first write and the timer will trigger after 1 ms.
+
+This allows the AsyncWrapper to handle high loads out of the box, but will also make it more aggressive on the
+target it is wrapping. Instead of writing 40000 msgs/sec then it will handle 400000 msgs/sec.
+
+### ColoredConsoleTarget supports Ansi output
+Console coloring now also works on VS Code or other console terminal using ANSI color escape sequences.
+
+```xml
+<target name="console" xsi:type="ColoredConsole" enableAnsiOutput="true" />
+```
+
+### AppSetting without NLog.Extended
+NLog.Extended was needed when wanting to use `${appsetting}` on .NET Framework. But it also had
+dependencies on System.Messaging (MSMQ). Since NLog already dependends on System.Configuration
+then it didn't make sense also to need NLog.Extended for reading app.config / web.config.
+
+### NLog configuration inside appsettings.json
```

Comment:
We need to merge https://github.com/NLog/NLog.Extensions.Logging/pull/263 first for this?

 (and mention NLog.Extensions.Logging 1.5?) - 

Doubting if this should be a separate post.

Reply:
You decide. There is no law saying that the news-posts should be posted on the same day as the release is cut.